langchain/prompts/chat.py

```diff
@@ -207,7 +223,40 @@ def partial(self, **kwargs: Union[str, Callable[[], str]]) -> BasePromptTemplate
 
     @property
     def _prompt_type(self) -> str:
-        raise NotImplementedError
+        """Return the prompt type key."""
+        return "chatPrompt"
+
+    def dict(self, **kwargs: Any) -> Dict:
+        prompt_dict = json.loads(self.json())
+
+        prompt_dict["_type"] = self._prompt_type
+        for i, message in enumerate(self.messages):
+            if isinstance(message, SystemMessagePromptTemplate):
+                prompt_dict["messages"][i]["prompt"]["role"] = "system"
```

Comment:
out of curiosity won't this already be in prompt_dict["messages"]["i"]? why does it need to be on "prompt"?